react/week4/githubsearcher/src/components/Search.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import TextField from '@material-ui/core/TextField';
+import { Context } from '../context';
+import Users from './User';
+import Loading from './Loading';
+
+class Search extends Component {
+   state = {
+      usersArray: [],
+      userName: '',
+   };
+
+   findUser = (e) => {
+      e.preventDefault();
+      fetch(`https://api.github.com/search/users?q=${this.state.userName}`)
+         .then((res) => res.json())
+         .then((data) => {
+            console.log(data.items);
+            this.setState({
+               usersArray: data.items,
+               userName: '',
```

Comment:
This is wrong. Your user list state should be kept inside the context. You're basically keeping 2 states in two different places. One inside the `Seach` and another one inside your context.
The state has to live inside the context and expose the proper methods for you to modify it.